components/nexus-core/src/main/java/org/sonatype/nexus/internal/orient/DatabaseServerImpl.java

```diff
@@ -147,6 +153,26 @@ private OServerConfiguration createConfiguration() {
 
     config.handlers = new ArrayList<>();
 
+    try {
+      // is hazelcast available? use reflection to avoid caching negative result
+      getClass().getClassLoader().loadClass("com.hazelcast.core.Hazelcast");
+
+      // configure Hazelcast handler if we are currently running in distributed mode
+      File etcDir = new File(applicationDirectories.getInstallDirectory(), "etc");
+      OServerHandlerConfiguration hazelcastHandler = new OServerHandlerConfiguration();
+      hazelcastHandler.clazz = SharedHazelcastPlugin.class.getName();
+      hazelcastHandler.parameters = new OServerParameterConfiguration[] {
+          new OServerParameterConfiguration("enabled", "true"),
+          new OServerParameterConfiguration("nodeName", localNodeAccess.getId()),
+          // HACK: This is apparently needed, have to understand where/why/how this is used
+          new OServerParameterConfiguration("configuration.db.default", new File(etcDir, "default-distributed-db-config.json").getCanonicalPath())
+      };
+      config.handlers.add(hazelcastHandler);
+    }
+    catch (final ClassNotFoundException | LinkageError e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)